src/Storages/Streaming/StorageRandom.cpp

```diff
@@ -416,8 +436,18 @@ class GenerateRandomSource final : public ISource
                 != ProtonConsts::RESERVED_COLUMN_NAMES.end();
             if (is_reserved_column || our_columns.hasDefault(elem.name))
                 continue;
+
             block_to_fill.insert(elem);
         }
+
+        Block block_to_fill_as_result(block_to_fill.cloneEmpty());
```

Comment:
Not need clone empty block_to_fill, since the `evaluateMissingDefaults` only requires first parameter is `const Block & header`.